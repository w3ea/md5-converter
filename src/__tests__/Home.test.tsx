import { render } from '@testing-library/react';
import Index from '../pages/index';

describe('<Index />', () => {
    it('should render home page elements', () => {
        const { getByText } = render(<Index />);

        expect(getByText('MD5 Converter')).toBeInTheDocument();
        expect(getByText('MD5')).toBeInTheDocument();
        expect(getByText('Welcome to MyMd5Converter. Use the form above to get started')).toBeInTheDocument();
        expect(getByText('5a1c94d0b5eb2d18ef2ebce8adaab77b')).toBeInTheDocument();
    });

    it('should match home page snapshot', () => {
        const { asFragment } = render(<Index />);
        expect(asFragment()).toMatchSnapshot();
    });
});