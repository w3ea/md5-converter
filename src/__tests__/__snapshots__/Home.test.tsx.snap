// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Index /> should match home page snapshot 1`] = `
<DocumentFragment>
  <div
    class="pt-2 md:pt-5 lg:pt-8 pb-20 flex flex-col space-y-18 md:space-y-20"
  >
    <form
      class="tw-form"
      data-testid="convert-form"
    >
      <h1
        class="mb-4 md:mb-5 text-5.5 md:text-6 font-bold text-gray-800 text-center"
      >
        MD5 Converter
      </h1>
      <div
        class="relative flex-grow"
      >
        <label
          class="text-3.75 text-gray-600 ml-1.5 leading-7"
          for="text"
        >
          Text
        </label>
        <textarea
          class="tw-form__text-field"
          data-testid="text"
          id="text"
          name="text"
        />
        <div
          class="tooltip"
        >
          <svg
            class="w-5 text-gray-400"
            fill="none"
            stroke="currentColor"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M8.228 9c.549-1.165 2.03-2 3.772-2 2.21 0 4 1.343 4 3 0 1.4-1.278 2.575-3.006 2.907-.542.104-.994.54-.994 1.093m0 3h.01M21 12a9 9 0 11-18 0 9 9 0 0118 0z"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
            />
          </svg>
          <p
            class="tooltip__text"
          >
            use Shift + Enter for new line
          </p>
        </div>
      </div>
      <div
        class="mt-3 flex md:justify-end"
      >
        <button
          class="tw-form__submit-button"
          data-testid="submit-button"
          type="submit"
        >
          <svg
            class="w-6"
            fill="none"
            stroke="currentColor"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M10.325 4.317c.426-1.756 2.924-1.756 3.35 0a1.724 1.724 0 002.573 1.066c1.543-.94 3.31.826 2.37 2.37a1.724 1.724 0 001.065 2.572c1.756.426 1.756 2.924 0 3.35a1.724 1.724 0 00-1.066 2.573c.94 1.543-.826 3.31-2.37 2.37a1.724 1.724 0 00-2.572 1.065c-.426 1.756-2.924 1.756-3.35 0a1.724 1.724 0 00-2.573-1.066c-1.543.94-3.31-.826-2.37-2.37a1.724 1.724 0 00-1.065-2.572c-1.756-.426-1.756-2.924 0-3.35a1.724 1.724 0 001.066-2.573c-.94-1.543.826-3.31 2.37-2.37.996.608 2.296.07 2.572-1.065z"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
            />
            <path
              d="M15 12a3 3 0 11-6 0 3 3 0 016 0z"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
            />
          </svg>
          <span
            class="text-4.25"
          >
            Convert
          </span>
        </button>
      </div>
    </form>
    <div
      class="tw-table"
    >
      <table
        class="table"
        summary="The table list of converted text contains the row number, text, md5, and action (delete item) columns"
      >
        <thead
          class="lg:bg-gray-800 lg:border-b lg:border-blue-100"
        >
          <tr
            class="hidden lg:flex w-full"
          >
            <th
              class="column__text"
            >
              Text
            </th>
            <th
              class="column__md5"
            >
              MD5
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td
              class="column column__text"
              data-label="Text"
            >
              <span>
                Welcome to MyMd5Converter. Use the form above to get started
              </span>
            </td>
            <td
              class="column column__md5"
              data-label="MD5"
            >
              <span>
                5a1c94d0b5eb2d18ef2ebce8adaab77b
              </span>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</DocumentFragment>
`;
